src/Symfony/Component/Lock/Store/InMemoryStore.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\SharedLockStoreInterface;
+
+/**
+ * InMemoryStore is a PersistingStoreInterface implementation using
+ * php-array to manage locks.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class InMemoryStore implements SharedLockStoreInterface
+{
+    private $locks = [];
+    private $readLocks = [];
+
+    public function save(Key $key)
+    {
+        $hashKey = (string) $key;
+        $token = $this->getUniqueToken($key);
+        if (isset($this->locks[$hashKey])) {
+            // already acquired
+            if ($this->locks[$hashKey] === $token) {
+                return;
+            }
+
+            // check for promotion
+            if (isset($this->readLocks[$hashKey][$token]) && 1 === \count($this->readLocks[$hashKey])) {
+                unset($this->readLocks[$hashKey]);
+                $this->locks[$hashKey] = $token;
+
+                return;
+            }
+
+            throw new LockConflictedException();
+        }
+
+        $this->locks[$hashKey] = $token;
+    }
+
+    public function saveRead(Key $key)
+    {
+        $hashKey = (string) $key;
+        $token = $this->getUniqueToken($key);
+
+        // check if lock is already acquired in read mode
+        if (isset($this->readLocks[$hashKey])) {
+            $this->readLocks[$hashKey][$token] = true;
+
+            return;
+        }
+
+        if (($this->locks[$hashKey] ?? $token) !== $token) {
+            throw new LockConflictedException();
+        }
+
+        $this->locks[$hashKey] = 'read';
```

Comment:
this line looks weird to me. Why do we need this ?

Reply:
indeed, this can be removed.